src/common/pty_bus.rs

```diff
@@ -169,31 +176,76 @@ impl PtyBus {
             send_plugin_instructions,
             receive_pty_instructions,
             os_input,
-            id_to_child_pid: HashMap::new(),
+            pids: HashMap::new(),
+            active_pane: None,
             debug_to_file,
             task_handles: HashMap::new(),
         }
     }
-    pub fn spawn_terminal(&mut self, file_to_open: Option<PathBuf>) -> RawFd {
-        let (pid_primary, pid_secondary): (RawFd, RawFd) =
-            self.os_input.spawn_terminal(file_to_open);
+    fn terminal_spawner(
```

Comment:
Do you think we can either give this a more descriptive name or merge it with `spawn_terminal` somehow? I find it a little hard to understand the difference between these two methods without being aware of the whole cwd system...